naming/src/main/java/com/alibaba/nacos/naming/push/v2/executor/PushExecutorRpcImpl.java

```diff
@@ -34,19 +39,26 @@
     
     private final RpcPushService pushService;
     
-    public PushExecutorRpcImpl(RpcPushService pushService) {
+    private final NamingMetadataManager metadataManager;
+    
+    public PushExecutorRpcImpl(RpcPushService pushService, NamingMetadataManager metadataManager) {
         this.pushService = pushService;
+        this.metadataManager = metadataManager;
```

Comment:
Also service metadata add into pushdata wrapper?

Reply:
I think PushdataWrapper only keep the data is better. metaDataManager like a global DB, if we put in this wrapper, user maybe not know update the metadataManager will change the global data.

Reply:
metadata and service both the relative data of push data. So saved in wrapper has no problem.

If you are afraid that the metadata or service will be changed in executors, you can clone a new one to cache.

If you input `metadataManager`, there may be two problem I has known.

1. In `PushExecutorDelegate`, we have an SPI for executors, if your push data not include metadata, but data selector need metdata, means all executors SPI must depend `metadataManager`, which may cause you change the executors SPI to add an method to inject `metadataManager`. Not a good design. 
2. For one push task, all subscribers should receive same data, if you input `metadataManager`, during the push task, the metadata changed, which will cause the push data maybe not same for one push task.

Reply:
Yes, agree. I will change it.